File: code_combat/4_Backwoods_Forest/160-Taunting/taunt.js
// Attack munchkins, call brawlers and ignore burls.

// This function defines the hero's behaviour about enemies.
function dealEnemy(enemy) {
    // If enemy.type is "munchkin":
    if (enemy.type == "munchkin") {
        // Then attack it:
        hero.attack(enemy);
    }
    // If the enemy's type is "brawler":
    if (enemy.type == "brawler") {
        // Then say something to call the brawler:
        hero.say("Bye Bye!");
    }
}

while (true) {
    var enemy = hero.findNearestEnemy();
    if (enemy) {
        dealEnemy(enemy);
    }
    else {
        hero.moveXY(30, 34);
    }
}

File: projects/code_combat/4_Backwoods_Forest/160-Taunting/taunt.py
def dealEnemy(enemy):
    if enemy.type == "munchkin":
        hero.attack(enemy)
    if enemy.type == "brawler":
        hero.say("Bye Bye!")

while True:
    enemy = hero.findNearestEnemy()
    if enemy:
        dealEnemy(enemy)
    else:
        hero.moveXY(30, 34)
